js/src/contract/item.ts

```diff
@@ -421,869 +41,283 @@ export class ItemProgram {
-              args.newItemMint,
-              args.originator ||
-                (this.program.provider as AnchorProvider).wallet.publicKey
-            )
-          )[0],
-        newItemTokenHolder:
-          accounts.newItemTokenHolder ||
-          args.originator ||
-          (this.program.provider as AnchorProvider).wallet.publicKey,
-        craftItemTokenAccountEscrow: craftItemEscrow,
-        craftItemTokenAccount,
-        craftItem: (
-          await getItemPDA(args.craftItemTokenMint, args.craftItemIndex)
-        )[0],
-        craftItemClass: (
-          await getItemPDA(args.craftItemClassMint, args.craftItemClassIndex)
-        )[0],
-        receiver: (this.program.provider as AnchorProvider).wallet.publicKey,
-        systemProgram: SystemProgram.programId,
-        tokenProgram: TOKEN_PROGRAM_ID,
-      })
-      .remainingAccounts(remainingAccounts)
-      .rpc();
+    args: ItemInstruction.RemoveCraftItemFromEscrowArgs,
+    accounts: ItemInstruction.RemoveCraftItemFromEscrowAccounts,
+    additionalArgs: ItemInstruction.RemoveCraftItemFromEscrowAdditionalArgs,
+    options?: SendOptions
+  ): Promise<{
+    txid: string;
+    slot: number;
+  }> {
+    const instruction = await this.instruction.removeCraftItemFromEscrow(
+      args,
+      accounts,
+      additionalArgs
+    );
+    return this.sendWithRetry(instruction, [], options);
   }
 
   async beginItemActivation(
-    args: BeginItemActivationArgs,
-    accounts: BeginItemActivationAccounts,
-    _additionalArgs: BeginItemActivationAdditionalArgs = {}
-  ) {
-    const remainingAccounts =
-      await generateRemainingAccountsGivenPermissivenessToUse({
-        permissivenessToUse: args.usagePermissivenessToUse,
-        tokenMint: accounts.itemMint,
-        parentMint: args.itemClassMint,
-        parentIndex: args.classIndex,
-        parent: (await getItemPDA(args.itemClassMint, args.classIndex))[0],
-        metadataUpdateAuthority: accounts.metadataUpdateAuthority,
-        program: this.program,
-      });
-
-    const itemClassKey = (
-      await getItemPDA(args.itemClassMint, args.classIndex)
-    )[0];
-
-    const [itemActivationMarker, itemActivationBump] =
-      await getItemActivationMarker({
-        itemMint: accounts.itemMint,
-        index: args.index,
-        usageIndex: new BN(args.usageIndex),
-        amount: args.amount,
-      });
-
-    const instructions = [],
-      signers = [];
+    args: Omit<ItemInstruction.BeginItemActivationArgs, "itemClass">,
+    accounts: ItemInstruction.BeginItemActivationAccounts,
+    additionalArgs: ItemInstruction.BeginItemActivationAdditionalArgs = {},
+    options?: SendOptions
+  ): Promise<{
+    txid: string;
+    slot: number;
+  }> {
+    const signers = [];
     const itemTransferAuthority =
-      accounts.itemTransferAuthority || web3.Keypair.generate();
-
+      accounts.itemTransferAuthority || web3.Keypair.generate(); // Pulling this up out of the instruction - is that ok?
```

Comment:
@long-banana This is another case where we need the keypair for the signers outside of the instruction

Reply:
Go ahead and handle it the same way you did the other one. I think it the keypair will be needed to sign the instruction you're returning, force it to be passed into the instruction accounts/args. That seems like a good pattern to use for any like this in the future.